Documentation/cmdref/cilium-agent.md

```diff
@@ -168,15 +168,15 @@ cilium-agent [flags]
       --ipsec-key-file string                                Path to IPSec key file
       --iptables-lock-timeout duration                       Time to pass to each iptables invocation to wait for xtables lock acquisition (default 5s)
       --iptables-random-fully                                Set iptables flag random-fully on masquerading rules
-      --ipv4-native-routing-cidr string                      Allows to explicitly specify the IPv4 CIDR for native routing. This value corresponds to the configured cluster-cidr.
+      --ipv4-native-routing-cidr string                      Allows to explicitly specify the IPv4 CIDR for native routing. When specified, Cilium assumes networking for this CIDR is preconfigured and hands traffic destined for that range to the Linux network stack without applying any SNAT. The network stack will then be responsible for routing it to its destination, potentially with any NAT applied if necessary. Moreover, if the Kubernetes cluster CIDR is included in the native routing CIDR, the user must configure the routes to reach pods, either manually or by setting the auto-direct-node-routes flag.
```

Comment:
> the user must configure the routes to reach pods, either manually or by setting the auto-direct-node-routes flag

I think it's still not 100%, as in the tunneling mode you don't need to manually configure routes to reach pods. Is ipv4-native-routing used in the tunneling mode?

Reply:
As far as I understand, typically if you're running tunneling mode, you wouldn't bother with ipv4-native-routing. Anything inside the cluster would be handled via tunnels, anything outside the cluster you would masquerade. Though in theory I guess you could enable both at the same time, and as long as you handle the appropriate routing to any destinations within the native routing CIDR range, it could work. Agree though, you wouldn't have to worry about pod ranges as Cilium will handle routing for this traffic.

Reply:
I think (haven't validated yet) that in the tunneling mode setting the ipv4-native-routing option can have the counterproductive effect. According to my old comment (https://github.com/cilium/cilium/pull/14611#issuecomment-760766052), in the tunnel mode pod2host doesn't go over the tunnel. So, setting the option would avoid SNAT-ing which could result in the packet (pod2host) drop if an underlying network drops packets from unknown subnets (the AWS EC2 default mode).

Reply:
:+1: I could see how that could be a problem yeah.

There's a problem with the other way though, if sending traffic from pod2host and the host has host firewall enabled, then if you NAT the traffic, the destination will see host2host traffic and the policy may allow pod2host because it thinks the traffic is actually host2host. Probably we need to take a closer look at the asymmetric tunnel forwarding issue and decide the right way to handle it for that case.

Reply:
How about, admitting that the semantics of above flag are not really clear, and short term use something like the following for the flag description:

> When specified, Cilium assumes networking for this CIDR is preconfigured and
> hands traffic destined for that range to the Linux network stack without
> applying any SNAT. Generally speaking, specifying a native routing CIDR
> implies that Cilium can depend on the underlying networking stack to route
> packets to their destination. To offer a concrete example, if Cilium is
> configured to use direct routing and the Kubernetes CIDR is included in the
> native routing CIDR, the user must configure the routes to reach pods, either
> manually or by setting the auto-direct-node-routes flag.